sftp/sftp.go

```diff
@@ -508,9 +508,9 @@ func (o *Object) Hash(r fs.HashType) (string, error) {
 	err = fs.ErrHashUnsupported
 	var outputBytes []byte
 	if r == fs.HashMD5 {
-		outputBytes, err = session.Output("md5sum " + o.path() + " | awk '{ print $1 }'")
+		outputBytes, err = session.Output("md5sum \"" + o.path() + "\" | awk '{ print $1 }'")
```

Comment:
Hm, if you mean by simple double quote escaping something like `"` → `\"`, I would assume `file $(rm -rf *) name` would still yield undesirable results.

Looking at other languages, this is how [Ruby does it](https://github.com/ruby/ruby/blob/trunk/lib/shellwords.rb#L139): 
```ruby
str.gsub(/([^A-Za-z0-9_\-.,:\/@\n])/, "\\\\\\1")
# A LF cannot be escaped with a backslash because a backslash + LF
# combo is regarded as a line continuation and simply ignored.
str.gsub!(/\n/, "'\n'")
```
and then insert it into the shell command *without* any double or single quotes. They claim it works for Bash and is compatible to some POSIX standard. I converted it into Go here: https://play.golang.org/p/uOOCC0aIv8 I guess `strings.Replace` could also be a Regex, but it's late 😄